src/main/java/MicroRaiden.java

```diff
@@ -85,8 +212,529 @@ public void listBalances(String account, String remotePeer) {
+    
+    private static String waitingForTransaction(String myTransactionID) {
+    	if(debugInfo) {
+    		System.out.println("Transaction ID = "+myTransactionID);
+    	}
+		boolean loop=true;
+		String blockNumber=new String();
+        while(loop){
+        	Object tempObj=null;
+        	try {
+                CloseableHttpClient httpClient = HttpClientBuilder.create().build();
+                HttpPost request = new HttpPost(rpcAddress);     
+                request.addHeader("content-type", "application/json");
+                String queryTransactionString = "{\"method\":\"eth_getTransactionReceipt\"," +
+                        "\"params\":[\"" +
+                        myTransactionID +
+                        "\"]," +
+                        "\"id\":42,\"jsonrpc\":\"2.0\"}";
+                StringEntity params = new StringEntity(queryTransactionString);
+                request.setEntity(params);
+                CloseableHttpResponse response = httpClient.execute(request);
+                String temp =new BasicResponseHandler().handleResponse(response);
+                response.close();
+                httpClient.close();
+              
+                JSONParser parser = new JSONParser();
+                JSONObject jobj=(JSONObject)parser.parse(temp);
+                for (Object key : jobj.keySet()) {
+                    if (((String)key).equalsIgnoreCase("result")) {
+                    	tempObj=jobj.get(key);
+                    }
+                }
+            }catch (UnsupportedEncodingException e) {
+            	System.out.println("UnsupportedEncodingException: "+e);
+            }catch (ClientProtocolException e) {
+            	System.out.println("ClientProtocolException: "+e);
+            }catch (IOException e) {
+            	System.out.println("IOException: "+e);
+            }catch(ParseException e) {
+            	System.out.println("ParseException: "+e);
+            }catch (NumberFormatException e){
+            	System.out.println("NumberFormatException="+e);
+            }
+            if(tempObj==null){
+                //do nothing
+            }else{
+            	loop=false;
+                JSONObject jsonObject=(JSONObject) tempObj;
+                //The jsonObject can be further parsed to get more information.
+                blockNumber = (String)jsonObject.get("blockNumber");
+            }
+            try {
+                Thread.sleep(INTERVAL_CHECK_TRANS_DONE);
+                System.out.print("\b\\");
+                Thread.sleep(INTERVAL_CHECK_TRANS_DONE);
+                System.out.print("\b|");
+                Thread.sleep(INTERVAL_CHECK_TRANS_DONE);
+                System.out.print("\b/");
+                Thread.sleep(INTERVAL_CHECK_TRANS_DONE);
+                System.out.print("\b-");
+
+            }catch (InterruptedException e) {
+            	
+            }
+        }
+        return blockNumber;
+	}
+    
+    public void getEtherBalance(String accountName){
```

Comment:
Might want to think about organizing things like this into a Wallet-like class or Ethereum class  (getting balances, reading account files, etc.) and then keep the Raiden class dealing strictly with interacting with the channels (and calling the functions from that class).